services/streammanager/kafka/kafkamanager.go

```diff
@@ -547,27 +549,31 @@ func (p *ProducerManager) Produce(jsonData json.RawMessage, destConfig interface
 		return makeErrorResponse(err) // returning 500 for retrying, in case of bad configuration
 	}
 
-	if conf.Topic == "" {
+	parsedJSON := gjson.ParseBytes(jsonData)
```

Comment:
Here, parsedJSON can be single message or batchMessage. For batchMessage, will topic be available at the outer json ?

Reply:
Yes for batch it will be available at outer json we will batch the messages based on topic